src/main/java/io/anserini/document/TweetDocument.java

```diff
@@ -72,79 +67,115 @@ public TweetDocument() {
   @Override
   public TweetDocument readNextRecord(BufferedReader reader) throws IOException {
     String line;
-    while ((line = reader.readLine()) != null) {
-      if(fromJson(line)) return this;
+    try {
+      while ((line = reader.readLine()) != null) {
+        if (fromJson(line)) {
+          return this;
+        } // else: not desired JSON data, read the next line
+      }
+    } catch (IOException e) {
+      LOG.error("Exception from BufferedReader:", e);
     }
     return null;
   }
 
   public boolean fromJson(String json) {
-    JsonObject obj = null;
+    ObjectMapper mapper = new ObjectMapper();
+    Status tweetObj = null;
     try {
-      obj = (JsonObject) JSON_PARSER.parse(json);
-    } catch (JsonSyntaxException e) {
+      tweetObj = mapper
+        .disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES) // Ignore unrecognized properties
+        .registerModule(new Jdk8Module()) // Deserialize Java 8 Optional: http://www.baeldung.com/jackson-optional
+        .readValue(json, Status.class);
+    } catch (IOException e) {
+//      LOG.error(e.getMessage());
```

Comment:
enable the log?

Reply:
There are too many such errors due to missing required JSON fields